specs/ExtendedProcessFailed.md

```diff
@@ -0,0 +1,671 @@
+[uuid(c62f5687-2f09-481b-b6e4-2c0620bc95a2), object, pointer_default(unique)]
+interface ICoreWebView2ProcessFailedEventArgs2 : IUnknown {
+  // The ProcessFailedKind property below is already in
+  // ICoreWebView2ProcessFailedEventArgs. The changes to this interface extend
+  // its return enum so it can now specify additional process kinds: gpu process
+  // exited, utility process exited, etc. The property is included here to show
+  // the updated docs.
+
+  /// The kind of process failure that has occurred. `processFailedKind` is
+  /// `COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_EXITED` if the
+  /// failed process is the main frame's renderer, even if there were subframes
+  /// rendered by such process; all frames are gone when this happens.
+  // [propget] HRESULT ProcessFailedKind(
+  //     [out, retval] COREWEBVIEW2_PROCESS_FAILED_KIND* processFailedKind);
+
+  /// The reason for the process failure. The reason is always
+  /// `COREWEBVIEW2_PROCESS_FAILED_REASON_UNEXPECTED` when `ProcessFailedKind`
+  /// is `COREWEBVIEW2_PROCESS_FAILED_KIND_BROWSER_PROCESS_EXITED`, and
+  /// `COREWEBVIEW2_PROCESS_FAILED_REASON_UNRESPONSIVE` when `ProcessFailedKind`
+  /// is `COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_UNRESPONSIVE`.
+  /// For other process failure kinds, the reason may be any of the reason
+  /// values.
+  [propget] HRESULT Reason(
+      [out, retval] COREWEBVIEW2_PROCESS_FAILED_REASON* reason);
+
+  /// The exit code of the failing process. The exit code is always `1` when
+  /// `ProcessFailedKind` is
+  /// `COREWEBVIEW2_PROCESS_FAILED_KIND_BROWSER_PROCESS_EXITED`, and
+  /// `STILL_ACTIVE` (`259`) when `ProcessFailedKind` is
+  /// `COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_UNRESPONSIVE`.
+  [propget] HRESULT ExitCode(
+      [out, retval] int* exitCode);
+
+  /// Description of the process assigned by the WebView2 Runtime. This is a
+  /// technical English term appropriate for logging or development purposes,
+  /// and not localized for the end user. It applies to utility processes (for example,
+  /// "Audio Service" or "Video Capture") and plugin processes (for example, "Flash").
+  /// The returned `processDescription` is `null` if the WebView2 Runtime did
+  /// not assign a description to the process.
+  [propget] HRESULT ProcessDescription(
+      [out, retval] LPWSTR* processDescription);
+
+  /// The list of frames in the `CoreWebView2` that were being rendered by the
+  /// failed process. The content in these frames is replaced with an error page.
+  /// This is only available when `ProcessFailedKind` is
+  /// `COREWEBVIEW2_PROCESS_FAILED_KIND_FRAME_RENDER_PROCESS_EXITED`;
+  /// `frames` is `null` for all other process failure kinds, including the case
+  /// in which the failed process was the renderer for the main frame and
+  /// subframes within it, for which the failure kind is
+  /// `COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_EXITED`.
+  [propget] HRESULT ImpactedFramesInfo(
```

Comment:
Should be ImpactedFrameInfos. "Infos" doesn't sound like a word, but this is a collection type, and "FramesInfo" is singular.